crates/manifest/src/schema/v2.rs

```diff
@@ -119,9 +119,26 @@ pub struct Component {
     pub build: Option<ComponentBuildConfig>,
 }
 
-/// Used to skip serializing if the value is either `None` or `Some(v)` where `v` is empty
-fn is_none_or_empty<T>(value: &Option<Vec<T>>) -> bool {
-    value.as_ref().map(|s| s.is_empty()).unwrap_or(true)
+impl Component {
+    /// Combine `allowed_outbound_hosts` with the deprecated `allowed_http_hosts` into
+    /// one array all normalized to the syntax of `allowed_outbound_hosts`.
+    pub fn normalized_allowed_outbound_hosts(&self) -> anyhow::Result<Vec<String>> {
+        let normalized =
+            crate::compat::convert_allowed_http_to_allowed_hosts(&self.allowed_http_hosts, false)?;
+        if !normalized.is_empty() {
+            terminal::warn!(
+                "Use of the deprecated field `allowed_http_hosts` - to fix, \
+            replace the use of `allowed_http_hosts` with `allowed_outbound_hosts = {normalized:?}`",
```

Comment:
suggestion: thoughts on reducing to just the following?

```suggestion
            replace with `allowed_outbound_hosts = {normalized:?}`",
```

Reply:
@vdice maybe 
```suggestion
            replace `allowed_http_hosts` with `allowed_outbound_hosts = {normalized:?}`",
```
so as to be absolutely specific?

Reply:
If there are no other changes, I might sneak this in to a follow-up PR to avoid having to run CI again, but otherwise 👍